### PR TITLE
prisma code models

### PR DESCRIPTION
**Pull request by paul-caron**

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -157,3 +158,49 @@ model Ban {
...
   note String?
 }
+
+
+
+
+model Code {
+  id               BigInt            @id @default(autoincrement())
+  createdAt        DateTime          @default(now())
+  updatedAt        DateTime?
```
Comment (DonDejvo) [inline backend/prisma/schema.prisma:168]: Use Prisma inbuilt @updatedAt

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -157,3 +158,49 @@ model Ban {
...
+  isPublic         Boolean           @default(false)
+  source           String            @db.VarChar(8192)
```
Comment (DonDejvo) [inline backend/prisma/schema.prisma:170]: Remove the varchar limit

File: backend/prisma/schema.prisma [hunk 3]
```diff
@@ -157,3 +158,49 @@ model Ban {
...
+  JS
+  C
+  LISP
+}
+
+model CodeVersion {
+  id               BigInt            @id @default(autoincrement())
+  createdAt        DateTime          @default(now())
+  name             String            @db.VarChar(64)
+  source           String            @db.VarChar
```
Comment (DonDejvo) [inline backend/prisma/schema.prisma:191]: @db.VarChar can be removed

File: backend/prisma/schema.prisma [hunk 4]
```diff
@@ -157,3 +158,49 @@ model Ban {
...
+  code             Code              @relation(fields:[codeId],references:[id], onDelete: Cascade)
+  codeId           BigInt            @unique
```
Comment (DonDejvo) [inline backend/prisma/schema.prisma:193]: Should not be unique

File: backend/prisma/schema2.prisma [hunk 5]
```diff
@@ -0,0 +1,144 @@
+// This is your Prisma schema file,
```
Comment (DonDejvo) [inline backend/prisma/schema2.prisma:1]: This file shouldn't be committed I guess it was for backup

File: backend/prisma/schema.prisma [hunk 6]
```diff
@@ -157,3 +158,49 @@ model Ban {
...
+}
+
+model CodeTemplate {
+  id               BigInt            @id @default (autoincrement())
+  language         CodeLanguage      @unique
+  source           String            @db.VarChar
```
Comment (DonDejvo) [inline backend/prisma/schema.prisma:199]: VarChar not required here

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (paul-caron) [reply backend/prisma/schema.prisma]: done

File: backend/prisma/schema.prisma (hunk 2 above)
Comment (paul-caron) [reply backend/prisma/schema.prisma]: done. removed the limit.

File: backend/prisma/schema.prisma (hunk 3 above)
Comment (paul-caron) [reply backend/prisma/schema.prisma]: done. removed.

File: backend/prisma/schema.prisma (hunk 4 above)
Comment (paul-caron) [reply backend/prisma/schema.prisma]: done.

File: backend/prisma/schema2.prisma (hunk 5 above)
Comment (paul-caron) [reply backend/prisma/schema2.prisma]: backup file removed.

File: backend/prisma/schema.prisma (hunk 6 above)
Comment (paul-caron) [reply backend/prisma/schema.prisma]: done. removed varchar.

**Pull request merged**